File: test_streaming.js
#!/usr/bin/env node

/**
 * Test script for LLM streaming WebSocket endpoint
 * Usage: node test_streaming.js [message] [language] [conversation_id]
 */

const WebSocket = require('ws');

const SERVER_URL = process.env.SERVER_URL || 'ws://localhost:8085';
const MESSAGE = process.argv[2] || 'Hello, how are you?';
const LANGUAGE = process.argv[3] || 'en_US';
const CONVERSATION_ID = process.argv[4] || undefined;

// Build WebSocket URL
let wsUrl = `${SERVER_URL}/ws/chat/stream?message=${encodeURIComponent(MESSAGE)}&language=${LANGUAGE}`;
if (CONVERSATION_ID) {
    wsUrl += `&conversation_id=${encodeURIComponent(CONVERSATION_ID)}`;
}

console.log(`Connecting to: ${wsUrl}`);
console.log(`Message: ${MESSAGE}`);
console.log(`Language: ${LANGUAGE}`);
if (CONVERSATION_ID) {
    console.log(`Conversation ID: ${CONVERSATION_ID}`);
}
console.log('---\n');

const ws = new WebSocket(wsUrl);

let tokenCount = 0;
let audioChunkCount = 0;
let fullText = '';
let startTime = Date.now();

ws.on('open', () => {
    console.log('✅ WebSocket connected\n');
    startTime = Date.now();
});

ws.on('message', (data) => {
    try {
        const message = JSON.parse(data.toString());
        
        switch (message.type) {
            case 'status':
                console.log(`📊 Status: ${message.status} - ${message.message || ''}`);
                if (message.status === 'complete') {
                    console.log(`\n📝 Full text: ${message.text || fullText}`);
                }
                break;
                
            case 'token':
                tokenCount++;
                fullText = message.text || fullText;
                process.stdout.write(message.token);
                break;
                
            case 'audio_chunk':
                audioChunkCount++;
                const audioSize = message.audio ? Math.round(message.audio.length / 1024) : 0;
                console.log(`\n🔊 Audio chunk #${audioChunkCount} (${audioSize}KB, ${message.sample_rate}Hz)`);
                break;
                
            case 'error':
                console.error(`\n❌ Error: ${message.error}`);
                break;
                
            default:
                console.log(`\n⚠️  Unknown message type: ${message.type}`, message);
        }
    } catch (error) {
        console.error(`\n❌ Error parsing message: ${error.message}`);
        console.error(`Raw data: ${data.toString().substring(0, 200)}`);
    }
});

ws.on('error', (error) => {
    console.error(`\n❌ WebSocket error: ${error.message}`);
});

ws.on('close', (code, reason) => {
    const duration = ((Date.now() - startTime) / 1000).toFixed(2);
    console.log(`\n---\n✅ Connection closed`);
    console.log(`   Code: ${code}`);
    console.log(`   Reason: ${reason.toString() || 'Normal closure'}`);
    console.log(`   Duration: ${duration}s`);
    console.log(`   Tokens received: ${tokenCount}`);
    console.log(`   Audio chunks received: ${audioChunkCount}`);
    console.log(`   Full text length: ${fullText.length} characters`);
    process.exit(0);
});

// Handle Ctrl+C
process.on('SIGINT', () => {
    console.log('\n\n⚠️  Interrupted by user');
    ws.close();
    process.exit(1);
});

// Timeout after 60 seconds
setTimeout(() => {
    console.log('\n\n⏱️  Timeout after 60 seconds');
    ws.close();
    process.exit(1);
}, 60000);

File: test_streaming.py
#!/usr/bin/env python3

"""
Test script for LLM streaming WebSocket endpoint
Usage: python3 test_streaming.py [message] [language] [conversation_id]
"""

import asyncio
import json
import sys
import time
import urllib.parse
from websockets.client import connect

SERVER_URL = "ws://localhost:8085"
MESSAGE = sys.argv[1] if len(sys.argv) > 1 else "Hello, how are you?"
LANGUAGE = sys.argv[2] if len(sys.argv) > 2 else "en_US"
CONVERSATION_ID = sys.argv[3] if len(sys.argv) > 3 else None

# Build WebSocket URL
params = {
    "message": MESSAGE,
    "language": LANGUAGE
}
if CONVERSATION_ID:
    params["conversation_id"] = CONVERSATION_ID

ws_url = f"{SERVER_URL}/ws/chat/stream?{urllib.parse.urlencode(params)}"

print(f"Connecting to: {ws_url}")
print(f"Message: {MESSAGE}")
print(f"Language: {LANGUAGE}")
if CONVERSATION_ID:
    print(f"Conversation ID: {CONVERSATION_ID}")
print("---\n")

token_count = 0
audio_chunk_count = 0
full_text = ""
start_time = None

async def test_streaming():
    global token_count, audio_chunk_count, full_text, start_time
    
    try:
        async with connect(ws_url) as websocket:
            print("✅ WebSocket connected\n")
            start_time = time.time()
            
            async for message in websocket:
                try:
                    data = json.loads(message)
                    
                    msg_type = data.get("type", "unknown")
                    
                    if msg_type == "status":
                        status = data.get("status", "")
                        msg = data.get("message", "")
                        print(f"📊 Status: {status} - {msg}")
                        if status == "complete":
                            text = data.get("text", full_text)
                            print(f"\n📝 Full text: {text}")
                    
                    elif msg_type == "token":
                        token_count += 1
                        token = data.get("token", "")
                        full_text = data.get("text", full_text)
                        print(token, end="", flush=True)
                    
                    elif msg_type == "audio_chunk":
                        audio_chunk_count += 1
                        audio_size = len(data.get("audio", "")) // 1024 if data.get("audio") else 0
                        sample_rate = data.get("sample_rate", 0)
                        print(f"\n🔊 Audio chunk #{audio_chunk_count} ({audio_size}KB, {sample_rate}Hz)")
                    
                    elif msg_type == "error":
                        print(f"\n❌ Error: {data.get('error', 'Unknown error')}")
                    
                    else:
                        print(f"\n⚠️  Unknown message type: {msg_type}", data)
                
                except json.JSONDecodeError as e:
                    print(f"\n❌ Error parsing message: {e}")
                    print(f"Raw data: {message[:200]}")
                except Exception as e:
                    print(f"\n❌ Error processing message: {e}")
    
    except Exception as e:
        print(f"\n❌ Connection error: {e}")
        return
    
    finally:
        if start_time:
            duration = time.time() - start_time
            print(f"\n---\n✅ Connection closed")
            print(f"   Duration: {duration:.2f}s")
            print(f"   Tokens received: {token_count}")
            print(f"   Audio chunks received: {audio_chunk_count}")
            print(f"   Full text length: {len(full_text)} characters")

if __name__ == "__main__":
    try:
        asyncio.run(test_streaming())
    except KeyboardInterrupt:
        print("\n\n⚠️  Interrupted by user")
        sys.exit(1)

